rvc_fastapi.py

```diff
@@ -0,0 +1,207 @@
+
+from typing import Union
+import os
+import sys
+from io import BytesIO
+
+from fastapi import HTTPException
+from fastapi.responses import StreamingResponse
+
+# don't like settings paths like this at all but due bad code its necessary
+now_dir = os.getcwd()
+sys.path.append(now_dir)
+print(now_dir)
+
+from dotenv import load_dotenv
+from scipy.io import wavfile
+
+from configs.config import Config
+from infer.modules.vc.modules import VC
+import fastapi
+import uvicorn
+
+# load_dotenv is also very bad practice but necessary due bad code
+load_dotenv()
+
+app = fastapi.FastAPI(
+    title="Retrieval-based Voice Conversion FastAPI",
+    summary="Infer previously cloned voices",
+    version="0.0.2",
+    contact={
+        "name": "w4hns1nn",
+        "url": "https://github.com/w4hns1nn",
+    }
+)
+
+tags = [
+    {
+        "name": "voice2voice",
+        "description": "Voice2Voice conversion using the pretrained model"
+    }
+]
+
+class ModelCache:
+    """
+    This class is used to cache the models so that they don't need to be loaded every time
+    """
+    def __init__(self):
+        self.models = {}
+
+    def load_model(self, model_name: str, device: str = None, is_half: bool = True):
+        if model_name not in self.models:
+            config = Config() # config_file_folder="A:/projects/Retrieval-based-Voice-Conversion-WebUI/configs/")
+            config.device = device if device else config.device
+            config.is_half = is_half if is_half else config.is_half
+            vc = VC(config)
+            vc.get_vc(f"{model_name}.pth")
+            self.models[model_name] = vc
+        return self.models[model_name]
+
+
+def infer(
+        input: Union[str, bytes], # filepath or raw bytes
+        model_name: str,
+        index_path: str = None,
+        f0up_key: int = 0,
+        f0method: str = "crepe",
+        index_rate: float = 0.66,
+        device: str = None,
+        is_half: bool = False,
+        filter_radius: int = 3,
+        resample_sr: int = 0,
+        rms_mix_rate: float = 1,
+        protect: float = 0.33,
+        **kwargs
+    ):
+    model_name = model_name.replace(".pth", "")
+
+    if index_path is None:
+        index_path = os.path.join("logs", model_name, f"added_IVF1254_Flat_nprobe_1_{model_name}_v2.index")
+        if not os.path.exists(index_path):
+            raise ValueError(f"autinferred index_path {index_path} does not exist. Please provide a valid index_path")
```

Comment:
Some models do not come with index files. If fails to find `index_path`, how about leave it to `None` instead of raising an Exception?

Reply:
The current version works without an index file just fine, just set "index_path": "None"